qiskit/transpiler/passes/scheduling/alignments/reschedule.py

```diff
@@ -104,69 +105,77 @@ def _push_node_back(self, dag: DAGCircuit, node: DAGOpNode, shift: int):
         Args:
             dag: DAG circuit to be rescheduled with constraints.
             node: Current node.
-            shift: Amount of required time shift.
         """
         node_start_time = self.property_set["node_start_time"]
         conditional_latency = self.property_set.get("conditional_latency", 0)
         clbit_write_latency = self.property_set.get("clbit_write_latency", 0)
 
-        nodes_with_overlap = [(node, shift)]
-        shift_stack = []
-        while nodes_with_overlap:
-            node, shift = nodes_with_overlap.pop()
-            shift_stack.append((node, shift))
-            # Compute shifted t1 of this node separately for qreg and creg
-            this_t0 = node_start_time[node]
-            new_t1q = this_t0 + node.op.duration + shift
-            this_qubits = set(node.qargs)
-            if isinstance(node.op, Measure):
-                # creg access ends at the end of instruction
-                new_t1c = new_t1q
-                this_clbits = set(node.cargs)
+        if isinstance(node.op, Gate):
+            alignment = self.pulse_align
+        elif isinstance(node.op, Measure):
+            alignment = self.acquire_align
+        else:
+            # Directive or delay. These can start at arbitrary time.
```

Comment:
I would be explicit about `isinstance(node.op, [Directive, Delay])` and leave the else for raising on an unknown op type, except that really means that all the non-gate/non-measure can start at arbitrary time.

Reply:
Done in [6d21886](https://github.com/Qiskit/qiskit-terra/pull/10077/commits/6d21886a1c6f0786bcba8c73e94e505cc19f0ef7).